src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -285,16 +287,28 @@ void LinearIR::unregister_expression(const ExpressionPtr& expr) {
         input->remove_consumer(expr->get_input_port(i));
     }
 
-    m_node2expression_map.erase(expr->get_node());
+    const auto& node = expr->get_node();
+    m_node2expression_map.erase(node);
+    if (ov::is_type<ov::op::v0::Parameter>(node)) {
+        auto it = std::find(m_parameter_expressions.cbegin(), m_parameter_expressions.cend(), expr);
+        OPENVINO_ASSERT(it != m_parameter_expressions.cend(),
+                        "unregister_expression is called for parameter expression which was not found in m_parameter_expressions");
+        m_parameter_expressions.erase(it);
+    } else if (ov::is_type<ov::op::v0::Result>(node)) {
+        auto it = std::find(m_result_expressions.cbegin(), m_result_expressions.cend(), expr);
+        OPENVINO_ASSERT(it != m_result_expressions.cend(),
+                        "unregister_expression is called for result expression which was not found in m_result_expressions");
+        m_result_expressions.erase(it);
+    }
 }
 
 LinearIR::exprIt LinearIR::insert(constExprIt pos, container::value_type&& value) {
-    register_expression(value);
+    register_expression(value, m_config.m_manual_build_support);
```

Comment:
As an idea: check `m_config.m_manual_build_support` value inside `register_expression` instead of setting it as parameter is everywhere,

Reply:
We have to call `register_expression(expr, true);` in two places: in constructor from `ov::Model` and in `clone()` method. So it seems like we can't check config value inside the registration